pkg/cmd/roachtest/run.go

```diff
@@ -209,6 +227,80 @@ func runTests(register func(registry.Registry), filter *registry.TestFilter) err
 	return err
 }
 
+// predictTestsToRun filters the tests to run based on certain criteria:
+// 1. the number of time a test has been successfully running
+// 2. the test is new
+// 3. the test has not been run for a while
+// 4. a subset of the successful tests
+func predictTestsToRun(
+	ctx context.Context, tests []registry.TestSpec, cloud, suite string,
+) ([]registry.TestSpec, int) {
+	options := []option.ClientOption{option.WithScopes(storage.ScopeReadOnly)}
+	cj := os.Getenv("GOOGLE_EPHEMERAL_CREDENTIALS")
+	if len(cj) != 0 {
+		options = append(options, option.WithCredentialsJSON([]byte(cj)))
+	} else {
+		fmt.Printf("GOOGLE_EPHEMERAL_CREDENTIALS env is not set.\n")
+	}
+	client, err := storage.NewClient(ctx, options...)
+	if err != nil {
+		fmt.Printf("connection to GCS failed: %v", err)
+		return tests, len(tests)
+	}
+	defer func() { _ = client.Close() }()
+
+	object := fmt.Sprintf("%s-%s-%s.%s", testsFileLocation, suite, cloud, testsCsvExtension)
+	r, err := client.Bucket(bucket).Object(object).NewReader(ctx)
+	if err != nil {
+		fmt.Printf("failed to get the object %s in bucket %s : %v\n", object, bucket, err)
+		return tests, len(tests)
+	}
+	body, err := io.ReadAll(r)
+	if err != nil {
+		fmt.Printf("failed to read CSV from GCS: %v", err)
+		return tests, len(tests)
+	}
+	if err = r.Close(); err != nil {
+		fmt.Printf("failed to close GCS writer: %v", err)
+		return tests, len(tests)
+	}
+	cr := csv.NewReader(bytes.NewReader(body))
+	data, err := cr.ReadAll()
+	if err != nil {
+		fmt.Printf("failed to read CSV data: %v", err)
+		return tests, len(tests)
+	}
+	testNamesToRun := make(map[string]string)
+	if len(data) <= 1 {
+		// the file has just the header
+		return tests, len(tests)
+	}
+	// csv columns:
+	// 0. TEST_NAME
+	// 1. SELECTED (yes/no)
+	// 2. AVG_DURATION
+	// 3. TOTAL_RUNS
+	for _, d := range data[1:] {
+		testNamesToRun[d[0]] = d[1]
+	}
+	selectedTestsCount := 0
+	for i := range tests {
+		if testShouldBeSkipped(testNamesToRun, tests[i]) {
+			tests[i].Skip = "stable test"
+			tests[i].SkipDetails = "test is stable. So, it is ignored."
```

Comment:
Maybe something like `test skipped because it is stable and predictive-tests is set`.